### PR TITLE
Add unit tests for squidpony.squidmath.NumberTools

### PR DESCRIPTION
**Pull request by sudo-jaa**
[generic PR text]

### CODE REVIEW

**Comment by petersear** [comment]
That's quite a lot of tests! Good ones too.
There are quite a few redundant tests which test the same thing as another with different values:
atan2_InputNegativeNegativeOutputPositive()
atan2_InputNegativePositiveOutputPositive2()
atan2_InputNegativePositiveOutputPositive3()
atan2_InputNegativePositiveOutputPositive4()

**Comment by sudo-jaa** [comment]
@petersear I seem to run into this duplicate test issue often, and i've never known really whether it is something we should be aiming to avoid. I hadn't removed them from here so far as I understand that it's testing different combinations of positive and negative numbers to achieve the same results. I'm not sure if this is redundant enough to warrant removing? 

@peterschrammel Do you have any insight into the matter?

**Comment by freddytuxworth** [comment]
`atan2` has at least 9 branches (with all those nested ternaries) so maybe these tests just cover every branch? Which seems like a good thing.

Other tests in this library use the many-asserts-in-one-method format, maybe that would make sense here?

**Comment by petersear** [comment]
Apologies for the confusion caused by copy and paste error. I had meant to list these functions ie. all tests providing the same type of input and not a mixture of negative and positive:
atan2_InputNegativeNegativeOutputPositive()
atan2_InputNegativeNegativeOutputPositive2()
atan2_InputNegativeNegativeOutputPositive3()
atan2_InputNegativeNegativeOutputPositive4()

**Comment by freddytuxworth** [comment]
The branching is not only dependent on the signs of the parameters though, also their magnitude and their difference, so those four tests might still be for different branches.

**Comment by petersear** [comment]
I've just checked those four tests and they do follow different paths. Two of these test the float version of the method and the other two test the double version. Within the pairs, they test different branches so I withdraw my "redundant" comment :-)

File: squidlib-util/src/test/java/squidpony/squidmath/NumberToolsTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package squidpony.squidmath;
+
+import org.junit.Assert;
+import org.junit.Test;
+import squidpony.squidmath.NumberTools;
+
+public class NumberToolsTest {
+
+  @Test
+  public void atan2_Test() {
```
Comment (freddytuxworth) [inline squidlib-util/src/test/java/squidpony/squidmath/NumberToolsTest.java]: Would recommend `testAtan2`, `testCos`, `testSway` style names in keeping with the existing tests.

**Review by petersear — APPROVED** [review]

**Comment by petersear** [comment]
@freddytuxworth see my earlier comment. You're right of course - I found that these tests were all valid.

File: squidlib-util/src/test/java/squidpony/squidmath/NumberToolsTest.java (hunk 1 above)
Comment (sudo-jaa) [reply squidlib-util/src/test/java/squidpony/squidmath/NumberToolsTest.java]: Resolved

**Review by freddytuxworth — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request closed without merging**